components/infobox/commons/infobox_basic.lua

```diff
@@ -0,0 +1,50 @@
+local Class = require('Module:Class')
+local Infobox = require('Module:Infobox')
+local getArgs = require('Module:Arguments').getArgs
+
+local BasicInfobox = Class.new(
+	function(self, frame)
+		local args = getArgs(frame)
+		self.frame = frame
+		self.pagename = mw.title.getCurrentTitle().title
+		self.name = args.name or self.pagename
+
+		if args.wiki == nil then
+			return error('Please provide a wiki!')
+		end
+
+		self.infobox = Infobox:create(frame, args.wiki)
+	end
+)
+
+--- Allows for overriding this functionality
+function BasicInfobox:getDisplayName(args)
+	return args.name
+end
```

Comment:
Could you explain why this is useful?

Reply:
it would be in combination with `BasicInfobox:addBaseDisplay`
because then we could overwrite the name display there
if we kick `BasicInfobox:addBaseDisplay` then we kick this function too